osu.Game/Rulesets/Mods/ModWindUp.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using osu.Framework.Audio;
+using osu.Framework.Audio.Track;
+using osu.Framework.Configuration;
+using osu.Framework.Timing;
+using osu.Framework.Graphics;
+using osu.Game.Beatmaps;
+using osu.Game.Graphics;
+using osu.Game.Rulesets.UI;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Objects.Types;
+
+namespace osu.Game.Rulesets.Mods
+{
+    public abstract class ModWindUp : Mod
+    {
+        public override string Name => "Wind Up";
+        public override string Acronym => "WU";
+        public override ModType Type => ModType.Fun;
+        public override FontAwesome Icon => FontAwesome.fa_chevron_circle_up;
+        public override string Description => "Crank it up!";
+        public override double ScoreMultiplier => 1;
+        public override Type[] IncompatibleMods => new[] { typeof(ModDoubleTime), typeof(ModHalfTime) };
+        public abstract double AppendRate { get; }
+    }
+
+    public abstract class ModWindUp<T> : ModWindUp, IUpdatableByPlayfield, IApplicableToClock, IApplicableToBeatmap<T>
```

Comment:
I'd move this to an abstract class called `TimeRampMod`, then have ModWindUp and ModWindDown derive from it.